src/plotApplications/components/infoCheck/PlotApplicationInfoCheckForm.js

```diff
@@ -0,0 +1,151 @@
+//@flow
+import React, {Component} from 'react';
+import {getFormValues, reduxForm} from "redux-form";
+import {Column, Row} from "react-foundation";
+import get from "lodash/get";
+import flowRight from 'lodash/flowRight';
+import {connect} from "react-redux";
+
+import {FieldTypes, FormNames} from "../../../enums";
+import {ButtonColors} from '$components/enums';
+import {getInfoCheckAttributes} from "../../selectors";
+import type {Attributes} from "../../../types";
+import FormField from "../../../components/form/FormField";
+import ModalButtonWrapper from "../../../components/modal/ModalButtonWrapper";
+import Button from "../../../components/button/Button";
+import {prepareInfoCheckForSubmission} from "../../helpers";
+import {getUserFullName} from "../../../users/helpers";
+
+type Props = {
+  onClose: Function,
+  onSubmit: Function,
+  attributes: Attributes,
+  valid: boolean,
+};
+
+class PlotApplicationInfoCheckForm extends Component<Props> {
+  firstField: ?React$Ref
+
+  componentDidMount(): void {
+    const infoCheck = this.props?.infoCheck;
+
+    if (!infoCheck) {
+      return;
+    }
+
+    const { preparer, ...rest } = infoCheck.data;
+
+    this.props.initialize({
+      ...rest,
+      preparer: preparer ? {
+        value: preparer.id,
+        label: getUserFullName(preparer)
+      } : null
+    });
+  }
+
+  setRefForFirstField = (element: React$Ref): void => {
+    this.firstField = element;
+  }
+
+  setFocus = (): void => {
+    if (this.firstField) {
+      this.firstField.focus();
+    }
+  }
+
+  handleSave = (): void => {
+    const {
+      onSubmit,
+    } = this.props;
+
+    onSubmit(prepareInfoCheckForSubmission(this.props.formValues));
+  };
+
+  render(): React$Node {
+    const {
+      attributes,
+      valid,
+      onClose
+    } = this.props;
+
+    return (
+      <form>
+        <Row>
+          <Column small={4}>
+            <FormField
+              setRefForField={this.setRefForFirstField}
+              fieldAttributes={get(attributes, 'preparer')}
```

Comment:
Other places in mvj-ui these field paths and titles (here: 'preparer') are stored as enums/constants.

Reply:
Weeell... yes and no. I feel there isn't much consistency there; in the plot search draft we built on for instance, that hadn't been done before, and the field names are still inlined to this day. I can do so here if you think it's useful, though.